### ISSUE TITLE
Mockito-Android 2.23.0 fails transformClassesWithDexBuilderForDevelopDebugAndroidTest

### ISSUE THREAD

**Issue by ZOlbrys** [mockito/mockito#1511]
I am running into the follow error when building and running my Android Instrumentation Tests after upgrading to Mockito-Android 2.23.0:

```
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:transformClassesWithDexBuilderForDevelopDebugAndroidTest'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:100)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
	at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:62)
	at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:60)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:97)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:87)
	at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:626)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:581)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.RuntimeException: com.android.build.api.transform.TransformException: com.android.builder.dexing.DexArchiveBuilderException: com.android.builder.dexing.DexArchiveBuilderException: Failed to process C:\Users\ZOlbr\.gradle\caches\modules-2\files-2.1\net.bytebuddy\byte-buddy\1.9.0\8cb0d5baae526c9df46ae17693bbba302640538b\byte-buddy-1.9.0.jar
	at com.android.builder.profile.Recorder$Block.handleException(Recorder.java:55)
	at com.android.builder.profile.ThreadRecorder.record(ThreadRecorder.java:104)
	at com.android.build.gradle.internal.pipeline.TransformTask.transform(TransformTask.java:212)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
	at org.gradle.api.internal.project.taskfactory.IncrementalTaskAction.doExecute(IncrementalTaskAction.java:46)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:121)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:110)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
	... 32 more
Caused by: com.android.build.api.transform.TransformException: com.android.builder.dexing.DexArchiveBuilderException: com.android.builder.dexing.DexArchiveBuilderException: Failed to process C:\Users\ZOlbr\.gradle\caches\modules-2\files-2.1\net.bytebuddy\byte-buddy\1.9.0\8cb0d5baae526c9df46ae17693bbba302640538b\byte-buddy-1.9.0.jar
	at com.android.build.gradle.internal.transforms.DexArchiveBuilderTransform.transform(DexArchiveBuilderTransform.java:329)
	at com.android.build.gradle.internal.pipeline.TransformTask$2.call(TransformTask.java:221)
	at com.android.build.gradle.internal.pipeline.TransformTask$2.call(TransformTask.java:217)
	at com.android.builder.profile.ThreadRecorder.record(ThreadRecorder.java:102)
	... 48 more
Caused by: com.android.builder.dexing.DexArchiveBuilderException: com.android.builder.dexing.DexArchiveBuilderException: Failed to process C:\Users\ZOlbr\.gradle\caches\modules-2\files-2.1\net.bytebuddy\byte-buddy\1.9.0\8cb0d5baae526c9df46ae17693bbba302640538b\byte-buddy-1.9.0.jar
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:593)
	at java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:677)
	at java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:720)
	at com.android.ide.common.internal.WaitableExecutor.waitForTasksWithQuickFail(WaitableExecutor.java:146)
	at com.android.build.gradle.internal.transforms.DexArchiveBuilderTransform.transform(DexArchiveBuilderTransform.java:308)
	... 51 more
Caused by: com.android.builder.dexing.DexArchiveBuilderException: Failed to process C:\Users\ZOlbr\.gradle\caches\modules-2\files-2.1\net.bytebuddy\byte-buddy\1.9.0\8cb0d5baae526c9df46ae17693bbba302640538b\byte-buddy-1.9.0.jar
	at com.android.build.gradle.internal.transforms.DexArchiveBuilderTransform.launchProcessing(DexArchiveBuilderTransform.java:805)
	at com.android.build.gradle.internal.transforms.DexArchiveBuilderTransform.lambda$convertToDexArchive$8(DexArchiveBuilderTransform.java:730)
	at java.util.concurrent.ForkJoinTask$AdaptedCallable.exec(ForkJoinTask.java:1424)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: com.android.builder.dexing.DexArchiveBuilderException: Error while dexing.
	at com.android.builder.dexing.D8DexArchiveBuilder.getExceptionToRethrow(D8DexArchiveBuilder.java:175)
	at com.android.builder.dexing.D8DexArchiveBuilder.convert(D8DexArchiveBuilder.java:152)
	at com.android.build.gradle.internal.transforms.DexArchiveBuilderTransform.launchProcessing(DexArchiveBuilderTransform.java:800)
	... 6 more
Caused by: com.android.tools.r8.CompilationFailedException: Compilation failed to complete
	at com.android.tools.r8.utils.ExceptionUtils.withCompilationHandler(ExceptionUtils.java:76)
	at com.android.tools.r8.utils.ExceptionUtils.withD8CompilationHandler(ExceptionUtils.java:45)
	at com.android.tools.r8.D8.run(D8.java:88)
	at com.android.builder.dexing.D8DexArchiveBuilder.convert(D8DexArchiveBuilder.java:147)
	... 7 more
Caused by: com.android.tools.r8.utils.AbortException
	at com.android.tools.r8.utils.Reporter.failIfPendingErrors(Reporter.java:77)
	at com.android.tools.r8.utils.Reporter.fatalError(Reporter.java:58)
	at com.android.tools.r8.utils.ExceptionUtils.withCompilationHandler(ExceptionUtils.java:67)
	... 10 more
```

Here is my gradle setup:

```
testImplementation 'org.mockito:mockito-core:2.23.0'
androidTestImplementation 'org.mockito:mockito-android:2.23.0'
```

If I downgrade to `org.mockito:mockito-android:2.22.0` the tests build and run fine.

I am experiencing this issue on Windows 10 and on Mac OS 10.13.6, both running Android Studio 3.1.4.

**Comment by raphw** [comment]
I assume that it's related to the module-info.class that us now included in Byte Buddy. Maybe one needs to set up the tooling to exclude it.

**Comment by bitsydarel** [comment]
Any news on this issue?
Experiencing the same issue.

**Comment by raphw** [comment]
I will probably release a new version of Byte Buddy where I use a multi-version jar and bundle the module-info.class in META-INF/versions/9.

This should however be fixed in the Android tool chain to begin with, those files will pop up more and more.

As a work-around, you can use Byte Buddy 1.8.22 as an explicit dependency in the mean-time.

**Comment by marcosalis** [comment]
I can reproduce this issue as well.

**Comment by darnmason** [comment]
Just to spell out the workaround:

    androidTestImplementation ('org.mockito:mockito-android:2.23.0') {
        exclude group: 'net.bytebuddy', module: 'byte-buddy'
        exclude group: 'net.bytebuddy', module: 'byte-buddy-android'
        exclude group: 'net.bytebuddy', module: 'byte-buddy-agent'
    }
    // Explicitly include byte-buddy 1.8.22 to fix a build error with Mockito 2.23
    androidTestImplementation 'net.bytebuddy:byte-buddy:1.8.22'
    androidTestImplementation 'net.bytebuddy:byte-buddy-android:1.8.22'
    androidTestImplementation 'net.bytebuddy:byte-buddy-agent:1.8.22'

**Comment by thevoiceless** [comment]
@darnmason Did that work for preventing `mockito-core` from pulling in the the newer version as well?

**Comment by darnmason** [comment]
@thevoiceless I only use `mockito-android` in my android test dependencies, which transitively pulls in `mockito-core`, and this works fine for me.

**Comment by francescocervone** [comment]
I don't know if it is related because the stacktrace is quite different, but looks like it is caused by Byte Buddy 1.9.x. Downgrading to 1.8.22 everything works fine. @raphw any news on this issue?

Below the stacktrace with Mockito `2.22.0` and Byte Buddy `1.9.3` (to prove that it's not Mockito `2.23.0`'s fault).
The same happens just by upgrading Mockito from `2.22.0` to `2.23.0`.
```
java.lang.UnsupportedOperationException: java.lang.UnsupportedOperationException
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:593)
        at java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:677)
        at java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:720)
        at com.android.ide.common.internal.WaitableExecutor.waitForTasksWithQuickFail(WaitableExecutor.java:146)
        at com.android.build.gradle.internal.transforms.DesugarIncrementalTransformHelper.getInitalGraphData(DesugarIncrementalTransformHelper.java:162)
        at com.android.build.gradle.internal.transforms.DesugarIncrementalTransformHelper.makeDesugaringGraph(DesugarIncrementalTransformHelper.java:130)
        at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:160)
        at com.android.build.gradle.internal.transforms.DesugarIncrementalTransformHelper.getDependenciesPaths(DesugarIncrementalTransformHelper.java:231)
        at com.android.build.gradle.internal.transforms.DexArchiveBuilderTransform.getD8DesugaringCacheInfo(DexArchiveBuilderTransform.java:461)
        at com.android.build.gradle.internal.transforms.DexArchiveBuilderTransform.transform(DexArchiveBuilderTransform.java:375)
        at com.android.build.gradle.internal.pipeline.TransformTask$2.call(TransformTask.java:239)
        at com.android.build.gradle.internal.pipeline.TransformTask$2.call(TransformTask.java:235)
        at com.android.builder.profile.ThreadRecorder.record(ThreadRecorder.java:102)
        at com.android.build.gradle.internal.pipeline.TransformTask.transform(TransformTask.java:230)
        at sun.reflect.GeneratedMethodAccessor520.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.IncrementalTaskAction.doExecute(IncrementalTaskAction.java:50)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipCachedTaskExecuter.execute(SkipCachedTaskExecuter.java:105)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ResolveBuildCacheKeyExecuter.execute(ResolveBuildCacheKeyExecuter.java:79)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.run(EventFiringTaskExecuter.java:51)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:46)
        at org.gradle.execution.taskgraph.LocalTaskInfoExecutor.execute(LocalTaskInfoExecutor.java:42)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:277)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:262)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:135)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:130)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.execute(DefaultTaskPlanExecutor.java:200)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.executeWithWork(DefaultTaskPlanExecutor.java:191)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.run(DefaultTaskPlanExecutor.java:130)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.UnsupportedOperationException
        at org.objectweb.asm.ClassVisitor.visitModule(ClassVisitor.java:135)
        at org.objectweb.asm.ClassReader.readModule(ClassReader.java:715)
        at org.objectweb.asm.ClassReader.accept(ClassReader.java:536)
        at org.objectweb.asm.ClassReader.accept(ClassReader.java:391)
        at com.android.builder.desugaring.DesugaringClassAnalyzer.analyze(DesugaringClassAnalyzer.java:144)
        at com.android.builder.desugaring.DesugaringClassAnalyzer.analyzeJar(DesugaringClassAnalyzer.java:92)
        at com.android.builder.desugaring.DesugaringClassAnalyzer.analyze(DesugaringClassAnalyzer.java:63)
        at com.android.build.gradle.internal.transforms.DesugarIncrementalTransformHelper.lambda$getInitalGraphData$4(DesugarIncrementalTransformHelper.java:150)
        at java.util.concurrent.ForkJoinTask$AdaptedCallable.exec(ForkJoinTask.java:1424)
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

**Comment by raphw** [comment]
Since 1.9.3, Byte Buddy contains module-info files, but they are hidden away as a multi-release jar file. It seems like Android does not properly handle either. The Android tool chain must learn how to deal with these concepts, as for the multi-release file, this behavior was never legal.

There is not much we can do on our side, Java has moved on a long while ago and it is up to Google to catch up.

**Comment by langerhans** [comment]
Related bug on the Android issue tracker: https://issuetracker.google.com/issues/77587908
TL;DR: This seems to be fixed in Android Gradle Plugin 3.3.0-alpha1 and is being considered as a backport to 3.2.2. I can't confirm the fix myself yet though.

**Comment by francescocervone** [comment]
I confirm that upgrading Android Gradle Plugin to 3.3.0-beta02 the build is successful.

**Comment by ZOlbrys** [comment]
This seems to be working in Android Studio 3.2.1 with Mockito 2.23.4.

**Comment by francescocervone** [comment]
@ZOlbrys in my case the issue still occurs. This means that they are different.
I'm going to create a separate issue.

**Comment by francescocervone** [comment]
Still doesn't work for me.

**Comment by LucianIacob** [comment]
> Just to spell out the workaround:
> 
> ```
> androidTestImplementation ('org.mockito:mockito-android:2.23.0') {
>     exclude group: 'net.bytebuddy', module: 'byte-buddy'
>     exclude group: 'net.bytebuddy', module: 'byte-buddy-android'
>     exclude group: 'net.bytebuddy', module: 'byte-buddy-agent'
> }
> // Explicitly include byte-buddy 1.8.22 to fix a build error with Mockito 2.23
> androidTestImplementation 'net.bytebuddy:byte-buddy:1.8.22'
> androidTestImplementation 'net.bytebuddy:byte-buddy-android:1.8.22'
> androidTestImplementation 'net.bytebuddy:byte-buddy-agent:1.8.22'
> ```

Worked for me. I will use this workaround until gradle build tools stable 3.3.0 version.

**Comment by RocketRider** [comment]
I upgraded to AGP 3.3 and Mockito 2.23.4 and now get a proguard warning.
Warning: class [META-INF/versions/9/module-info.class] unexpectedly contains class [module-info]

**Comment by TimvdLippe** [comment]
@RocketRider That is an issue with ProGuard not understanding multi-release jars and is unrelated to Mockito. For more information see https://openjdk.java.net/jeps/238

**Comment by krazykira** [comment]
Just wondering but does this issue only occur on a multiModule project because it didn't happen for my simple project with 1 module ?

**Comment by MatFl** [comment]
This happens when using android test modules.

**Comment by krazykira** [comment]
For curiosity i created a sample project with multiple modules and a lot of dependencies and i was still not able to get this error. I am wondering why is that. 


**app/build.gradle** test and instrumented test dependencies
```
    testImplementation 'junit:junit:4.12'
    // required if you want to use Mockito for unit tests
    testImplementation 'org.mockito:mockito-core:2.23.4'
    testImplementation 'com.nhaarman.mockitokotlin2:mockito-kotlin:2.1.0'
    // required if you want to use Mockito for Android tests
    androidTestImplementation 'org.mockito:mockito-android:2.23.4'
    // expresso related dependencies
    androidTestImplementation('com.android.support.test:runner:1.0.2') {
        exclude group: 'com.android.support'
    }
    androidTestImplementation 'com.android.support.test:rules:1.0.2'
    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
    androidTestImplementation 'com.android.support.test.espresso:espresso-contrib:3.0.2'
    androidTestImplementation 'com.android.support.test.espresso:espresso-intents:3.0.2'
    // required for replacing dependenices inside the app with mocked dependencies when needed
    androidTestImplementation 'com.github.fabioCollini.daggermock:daggermock-kotlin:0.8.4'
    androidTestImplementation 'com.github.fabioCollini.daggermock:daggermock:0.8.4'
    // required for opening final classes for expresso tests
    androidTestImplementation 'com.github.tmurakami:dexopener:1.0.5'
    // required for generating dagger graph using test dependencies
    kaptAndroidTest 'com.google.dagger:dagger-compiler:2.16'
    androidTestImplementation 'com.nhaarman.mockitokotlin2:mockito-kotlin:2.1.0'
```

**project/build.gradle**
```
// Top-level build file where you can add configuration options common to all sub-projects/modules.

buildscript {
    ext.kotlin_version = '1.3.20'
    repositories {
        google()
        jcenter()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.2.1'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}

allprojects {
    repositories {
        google()
        jcenter()
        maven { url "https://jitpack.io" }
    }
}
task clean(type: Delete) {
    delete rootProject.buildDir
}

```

**Comment by menjoo** [comment]
I was able to solve this by adding the following in the root build.gradle (since i have a multi module project)

```
subprojects {
    project.configurations.all {
        resolutionStrategy.eachDependency { details ->
            if (details.requested.group == 'net.bytebuddy') {
                details.useVersion "1.8.22"
            }
        }
    }
}
```

**Comment by chrisjenx** [comment]
@krazykira seems to only happen on multi module projects

**Comment by chrisjenx** [comment]
Is there no fix in the works for this?

**Comment by thevoiceless** [comment]
I no longer see this issue when using Mockito 2.24.0. Everything works without pinning the ByteBuddy version.

**Comment by chrisjenx** [comment]
I had to down grade from 24. Guess what ever breaks for me is different for you.

**Comment by TimvdLippe** [comment]
For reference, this is the Proguard bug: https://sourceforge.net/p/proguard/bugs/671/

**Comment by chrisjenx** [comment]
After upgrading to Android Gradle Plugin 3.3.2 and updating to 2.25 all seems fine now.. I think jetifier was the culprit as that was throwing an error in some versions.

**Comment by dapengzhang0** [comment]
I still have problem with Android Gradle Plugin 3.3.2 and mockito 2.25.1,
```
gradlew app:transformClassesAndResourcesWithProguardForRelease
```
fails. (See the above ref)

**Comment by TimvdLippe** [comment]
That's the Proguard bug I was talking about in https://github.com/mockito/mockito/issues/1511#issuecomment-475673168

**Comment by zawadz88** [comment]
By applying the fix suggested in https://github.com/mockito/mockito/issues/1511#issuecomment-475673168 I got it working in my Android project.
So in my `build.gradle` I had to add:
```
android {
    defaultConfig {
        packagingOptions {
            exclude 'META-INF/versions/9/module-info.class'
        }
    }
}
```
this got rid of `transformClassesAndResourcesWithProguard` issue for me.

However, in the project I'm working on we also use Firebase Performance Plugin and there was a similar issue there:
`META-INF/versions/9/module-info.class: D8: Unsupported source file type`

I got rid of it by adding:
```
tasks.whenTaskAdded { task ->
    if (task.name.contains('transformClassesWithFirebasePerformancePlugin') && task.name.contains('AndroidTest')) {
        task.enabled = false
    }
}
```
as we don't really need that to be applied to our test app.

**Comment by gummibjorn** [comment]
@zawadz88 ``META-INF/versions/9/module-info.class: D8: Unsupported source file type`` is a known issue in Android Studio and is going to be fixed in the next version.

https://issuetracker.google.com/issues/131114388

**Comment by menshen** [comment]
> Just to spell out the workaround:
> 
> ```
> androidTestImplementation ('org.mockito:mockito-android:2.23.0') {
>     exclude group: 'net.bytebuddy', module: 'byte-buddy'
>     exclude group: 'net.bytebuddy', module: 'byte-buddy-android'
>     exclude group: 'net.bytebuddy', module: 'byte-buddy-agent'
> }
> // Explicitly include byte-buddy 1.8.22 to fix a build error with Mockito 2.23
> androidTestImplementation 'net.bytebuddy:byte-buddy:1.8.22'
> androidTestImplementation 'net.bytebuddy:byte-buddy-android:1.8.22'
> androidTestImplementation 'net.bytebuddy:byte-buddy-agent:1.8.22'
> ```

this is worked for me , my error message is
`Illegal class file: Class module-info is missing a super type.`
